### PR TITLE
Run once at start balance

### LINKED ISSUE #1222: Can't access balance on "Run once at start"
Hey, I'm also having a problem that i belive it might be a bug...
I can't acess my balance on "run once at start" 
In this case I'm trying set the stake to 3% of my balance, but it always says it needs to be a positive number..
If i had it on the "before purchasing" it works fine..

_Originally posted by @mexarrinha in https://github.com/binary-com/binary-bot/issues/1216#issuecomment-463692118_

### PR DESCRIPTION
**Pull request by aaron-binary**
Fixes #1222, #1830

### CODE REVIEW

**Comment by esanshogo** [comment]
Hello guys please i need a real working bot that always win and make profit supporting xml file please anyone should kindly send it to this esanshogo@gmail.com  thanks...

**Comment by esanshogo** [comment]
Please help me with the good bot that support xml file    thank you

File: src/botPage/bot/TradeEngine/index.js [hunk 1]
```diff
@@ -118,8 +118,11 @@ export default class TradeEngine extends Balance(Purchase(Sell(OpenContract(Prop
 
                 // Only subscribe to balance in browser, not for tests.
                 if (document) {
-                    this.api.subscribeToBalance().then(r => {
-                        this.balance = Number(r.balance.balance);
+                    this.api.subscribeToBalance().then(response => {
+                        globalObserver.setState({
```
Comment (sam-binary) [inline src/botPage/bot/TradeEngine/index.js]: can you please destruct `response` ?

`const {balance: {balance, currency}} = response`

**Comment by esanshogo** [comment]
hello,please help me with the bot here xml file please, thank you.

**Pull request merged**